src/main/java/org/apache/accumulo/examples/helloworld/ReadData.java

```diff
@@ -27,52 +26,29 @@
 import org.apache.accumulo.core.data.Key;
 import org.apache.accumulo.core.data.Range;
 import org.apache.accumulo.core.data.Value;
-import org.apache.accumulo.examples.cli.ClientOnRequiredTable;
-import org.apache.accumulo.examples.cli.ScannerOpts;
+import org.apache.accumulo.core.security.Authorizations;
 import org.apache.hadoop.io.Text;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.beust.jcommander.Parameter;
-
 /**
- * Reads all data between two rows; all data after a given row; or all data in a table, depending on the number of arguments given.
+ * Reads all data between two rows
  */
 public class ReadData {
 
   private static final Logger log = LoggerFactory.getLogger(ReadData.class);
 
-  static class Opts extends ClientOnRequiredTable {
-    @Parameter(names = "--startKey")
-    String startKey;
-    @Parameter(names = "--endKey")
-    String endKey;
-  }
-
   public static void main(String[] args) throws AccumuloException, AccumuloSecurityException, TableNotFoundException {
-    Opts opts = new Opts();
-    ScannerOpts scanOpts = new ScannerOpts();
-    opts.parseArgs(ReadData.class.getName(), args, scanOpts);
-
-    Connector connector = opts.getConnector();
 
-    Scanner scan = connector.createScanner(opts.getTableName(), opts.auths);
-    scan.setBatchSize(scanOpts.scanBatchSize);
-    Key start = null;
-    if (opts.startKey != null)
-      start = new Key(new Text(opts.startKey));
-    Key end = null;
-    if (opts.endKey != null)
-      end = new Key(new Text(opts.endKey));
-    scan.setRange(new Range(start, end));
-    Iterator<Entry<Key,Value>> iter = scan.iterator();
+    Connector connector = Connector.builder().usingProperties("conf/accumulo-client.properties").build();
```

Comment:
When I first saw this I was thinking "maybe the props file could be args[0]".  However I convinced myself that the literal string is better because its easier to understand.

Reply:
It would also make argument parsing tricky.